source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -120,6 +121,18 @@ AWSLambdaFilter::decodeHeaders(Http::RequestHeaderMap &headers,
   return Http::FilterHeadersStatus::StopIteration;
 }
 
+
+Http::FilterHeadersStatus 
+AWSLambdaFilter::encodeHeaders(Http::ResponseHeaderMap &headers, bool ) {
+
+  if (!headers.get(AWSLambdaHeaderNames::get().FunctionError).empty()){
+    // We treat upstream errors as if it was any other upstream error
+    headers.setStatus(504);
+    return Http::FilterHeadersStatus::Continue;
+  }
+  return Http::FilterHeadersStatus::StopIteration;
```

Comment:
Why are we stopping iteration here?

Reply:
i think we should always return continue here`

Reply:
Ah I thought I changed that good catch.

Reply:
i think we should always continue here

Reply:
Updated to always send continue and have that continue out of the if.